### PR TITLE
Add the WAST library for testing

### PR DESCRIPTION
**Pull request by wycst**
Hello, this open-source performance testing project is fantastic. Can we add 'wast'（ https://github.com/wycst/wast ）Participate in performance testing, thank you very much!

### CODE REVIEW

File: src/main/java/com/github/fabienrenaud/jjb/support/Library.java [hunk 1]
```diff
@@ -35,7 +35,9 @@ public enum Library {
     UNDERSCORE_JAVA,
     PUREJSON,
     ANTONS,
-    QUICKBUF_JSON;
+    QUICKBUF_JSON,
+    WAST
+    ;
```
Comment (fabienrenaud) [inline src/main/java/com/github/fabienrenaud/jjb/support/Library.java:40]: Move semi-colon to the same line as WAST.

**Comment by fabienrenaud** [comment]
Yes, on the condition that performance is higher than a few of the lowest performers in the benchmark. Please run ser+deser  benchmarks for jackson,wast,gson.

File: src/main/java/com/github/fabienrenaud/jjb/support/Library.java (hunk 1 above)
Comment (wycst) [reply src/main/java/com/github/fabienrenaud/jjb/support/Library.java]: Okay, it has been adjusted

**Comment by wycst** [comment]
> Yes, on the condition that performance is higher than a few of the lowest performers in the benchmark. Please run ser+deser benchmarks for jackson,wast,gson.

The performance should be faster than most libraries, including the top few libraries, which I only tested on the window platform

**Comment by fabienrenaud** [comment]
I’m sure it’ll be fine.
Please post the results of the benchmark here.

Fabien

**Comment by wycst** [comment]
Hello, compared to the results of the first two high-performance library tests, the results of the just tested run

fastjson2:  2.0.47
dsl.version 2.0.2
wast: 0.0.12.1
~~~
# JMH version: 1.36
# VM version: JDK 17.0.3.1, Java HotSpot(TM) 64-Bit Server VM, 17.0.3.1+2-LTS-6
# VM invoker: D:\Program Files\Java\jdk-17.0.3.1\bin\java.exe
# VM options: --add-opens=java.base/java.time=ALL-UNNAMED --add-modules=jdk.incubator.vector -javaagent:D:\Program Files\JetBrains\IntelliJ IDEA 2023.1\lib\idea_rt.jar=62683:D:\Program Files\JetBrains\IntelliJ IDEA 2023.1\bin -Dfile.encoding=UTF-8

Benchmark                                     Mode  Cnt        Score        Error  Units
JavaJsonJmhDeserTest.dsljson_deser_client    thrpt    5   631549.154 ±  11858.434  ops/s
JavaJsonJmhDeserTest.fastjson2_deser_client  thrpt    5   680066.404 ±  36023.151  ops/s
JavaJsonJmhDeserTest.wast_deser_client       thrpt    5   905647.903 ±  25700.604  ops/s

JavaJsonJmhSerTest.dsljson_ser_client        thrpt    5  1109871.104 ± 107896.167  ops/s
JavaJsonJmhSerTest.fastjson2_ser_client      thrpt    5  1414414.026 ±  30514.291  ops/s
JavaJsonJmhSerTest.wast_ser_client           thrpt    5  1441172.972 ±  64336.484  ops/s
~~~
The original test data can be found in [README_2024_04_17_JSON_JMH_RAW.md](https://github.com/wycst/wast-jmh-test/blob/main/README_2024_04_17_JSON_JMH_RAW.md)

**Review by fabienrenaud — APPROVED** [review]

**Comment by wycst** [comment]
> Please also update the README with the new lib name and version.

I just updated the new library and version, and I found that the versions of other libraries are not the latest, such as dsl json and fastjson2

**Comment by wycst** [comment]
> Impressive! I wonder how you got serialization running 50% faster than fastjson. That's huge!



> Impressive! I wonder how you got serialization running 50% faster than fastjson. That's huge!

It's not that exaggerated, serialization is similar, but deserialization is faster than fastjson2, without using any black technology, only using unsafe and cached reflection. Most of the performance comes from the performance advantages of higher versions of JDK for reading and writing Strings, such as some APIs that add @ IntrinsicCandidate annotations, which cannot be surpassed by any coding techniques.

**Review by fabienrenaud — APPROVED** [review]

**Comment by fabienrenaud** [comment]
Thank you for your contribution!

**Pull request merged**